preseed/provision.erb

```diff
@@ -131,8 +105,8 @@ d-i apt-setup/local<%= repos %>/key string http://keyserver.ubuntu.com/pks/looku
 # Install minimal task set (see tasksel --task-packages minimal)
 tasksel tasksel/first multiselect minimal, ssh-server, openssh-server
 
-<% if puppet_enabled %>
-<% puppet_package = @host.param_true?('enable-puppetlabs-pc1-repo') ? 'puppet-agent' : 'puppet' -%>
+<% if puppet_enabled && !(@host.param_true?('enable-puppetlabs-repo') || @host.param_true?('enable-puppetlabs-pc1-repo')) %>
+<% puppet_package = 'puppet' -%>
```

Comment:
Does that mean that on older distros that ship puppet 3 by default it's now no longer possible to install puppet 4?